### PR TITLE
Return the default resolver if the factory URL refers directly to a devfile

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Check if the factory URL is a raw devfile URL. If so, use the default resolver,  which resolves factories from a direct URL to a devfile content.

### What issues does this PR fix or reference?

https://issues.redhat.com/browse/CRW-4366

### How to test this PR?

1. Deploy Che using the PR image: `chectl server:deploy -p=openshift -i=quay.io/ivinokur/che-server:next`
2. Start a workspace from a gitlab / bitbucket RAW devfile URL.
See workspace starts successfully.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java [hunk 1]
```diff
@@ -185,6 +188,14 @@ protected static class FactoryParametersResolverHolder {
      */
     public FactoryParametersResolver getFactoryParametersResolver(Map<String, String> parameters)
         throws BadRequestException {
+      String url = parameters.get(URL_PARAMETER_NAME);
+      // Check if the URL is a raw devfile URL. If so, use the default resolver,
+      // which resolves factories from a direct URL to a devfile content.
+      if (!isNullOrEmpty(url)
+          && devfileFilenamesProvider.getConfiguredDevfileFilenames().stream()
```
Comment (tolusha) [inline wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java:195]: Why `defaultFactoryResolver.accept(parameters)` doesn't work here?

Comment (vinokurig) [reply wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java:195]: Moving the `devfileFilenamesProvider.getConfiguredDevfileFilenames().stream().anyMatch(url::endsWith)` check to the `defaultFactoryResolver.accept(parameters)` check.

Comment (vinokurig) [reply wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java]: @tolusha Moved the logic to the `defaultFactoryResolver.accept(parameters)` please take a look.

**Comment by vinokurig** [comment]
@tolusha renamed the `DefaultFactoryParameterResolver` class to `RawDevfileUrlFactoryParameterResolver` please take a look.

**Review by tolusha — APPROVED** [review]

**Comment by ibuziuk** [comment]
@vinokurig could you please cherry-pick it to 7.67 for Dev Spaces 3.7?

**Pull request merged**